src/orchestrator.ts

```diff
@@ -186,17 +187,36 @@ export class Orchestrator {
             }
         } catch (error) {
             log(`Error: ${error}`);
-            context.done(
-                error,
-                new OrchestratorState({
-                    isDone: false,
-                    output: undefined,
-                    actions,
-                    error: error.stack,
-                    customStatus: this.customStatus,
-                }),
-            );
-            return;
+            if (error instanceof DurableFailure) {
+                // We will return the function execution as a success. It is up to the C# shim
+                // to correctly process the error only after reprocessing all of the actions to
+                // prevent a non-determinism error.
+                context.done(
+                    undefined,
```

Comment:
Unfortunately, I do believe this would result in a monitoring difference. I would have to double check with @mhoeger about the implications of this, and if there is any other workaround.

Right now, the problem is that failing the function execution means we get an RPC exception, with no useful information to perform the replay on the Durable Extension side of things. 

One potential workaround is to re-execute the orchestration function one more time from the C# extension if we detect a failure. This time we could indicate that the execution should fail when it encounters a `DurableFailure`. Note that this would also result in a monitoring difference between C# and JS, but at least customers would see an orchestration failure in App Insights. It also has a slight cost implication, due to the extra function execution.

Reply:
So I think I found a workaround. The RPC exception actually does have access to the Error returned (at least the message and the stacktrace). I could theoretically embed the `OrchestratorState` object in the error message, and then the C# shim could parse out the object from the exception. The downside is that this would require customers to upgrade both their DurableJS version as well as their extension version to get the fixed behavior.

Reply:
I should have addressed this in my latest iteration.